When a port is detected/configured as router port, dont process
queries/leaves on that port

diff --git a/net/bridge/br_multicast.c b/net/bridge/br_multicast.c
index b16f5cd..fb59cd3 100644
--- a/net/bridge/br_multicast.c
+++ b/net/bridge/br_multicast.c
@@ -731,7 +731,8 @@ static int br_multicast_add_group(struct net_bridge *br,
 
 	spin_lock(&br->multicast_lock);
 	if (!netif_running(br->dev) ||
-	    (port && port->state == BR_STATE_DISABLED))
+	    (port && (port->state == BR_STATE_DISABLED ||
+				!hlist_unhashed(&port->rlist))))
 		goto out;
 
 	mp = br_multicast_new_group(br, port, group);
@@ -1431,7 +1432,8 @@ static void br_multicast_leave_group(struct net_bridge *br,
 
 	spin_lock(&br->multicast_lock);
 	if (!netif_running(br->dev) ||
-	    (port && port->state == BR_STATE_DISABLED))
+	    (port && (port->state == BR_STATE_DISABLED ||
+				!hlist_unhashed(&port->rlist))))
 		goto out;
 
 	mdb = mlock_dereference(br->mdb, br);
